build.sh

```diff
@@ -178,6 +184,49 @@ setup_build_environment() {
   BINDIR="${BINROOT}/${FULL_VARIANT}/${OUTDIR}"
 }
 
+#-----------------------------------------------------------------------------
+# Function: prepare_coverage_capture
+# Run lcov preparations before testing for coverage
+#-----------------------------------------------------------------------------
+prepare_coverage_capture() {
+  [[ -n $GITHUB_ACTIONS ]] && echo "::group::Code Coverage Preparation"
+  lcov --zerocounters      --directory $BINROOT --base-directory $ROOT
+  lcov --capture --initial --directory $BINROOT --base-directory $ROOT -o $BINROOT/base.info
+  [[ -n $GITHUB_ACTIONS ]] && echo "::endgroup::"
+}
+
+#-----------------------------------------------------------------------------
+# Function: capture_coverage
+# Capture coverage collected since `prepare_coverage_capture` was invoked
+#-----------------------------------------------------------------------------
+capture_coverage() {
+  NAME=${1:-cov} # Get output name. Defaults to `cov.info`
+
+  [[ -n $GITHUB_ACTIONS ]] && echo "::group::Code Coverage Capture ($NAME)"
+
+  # Capture coverage collected while running tests previously
+  lcov --capture --directory $BINROOT --base-directory $ROOT -o $BINROOT/test.info
+
+  # Accumulate results with the baseline captured before the test
+  lcov --add-tracefile $BINROOT/base.info --add-tracefile $BINROOT/test.info -o $BINROOT/full.info
+
+  # Extract only the coverage of the project source files
+  lcov --output-file $BINROOT/$NAME.info --extract $BINROOT/full.info \
+    "$ROOT/src/*" \
+    "$ROOT/deps/triemap/*" \
+    "$ROOT/deps/thpool/*" \
```

Comment:
I suggest covering all the deps that are not GitHub submodules
So we can see in front oof ur eyes which code we imported for no reason and clean it up later on

Reply:
This is the right way to work. First, describe which directories are needed for coverage, and then remove specific sub-directories that are not required.
There is no benefit for including other `deps` we don't care about. Until this PR, we didn't have any coverage report of code under `deps`, so at least we have it now for owned code